Include error indicator field in HWP ManifestDb

yukisakurai opened the pull request:
Added a field (`hwp_err`) in HWP ManifestDb to indicate when the HWP angle is all zeros, whether it is a calculation failure or just a lack of L2 HWP data.
Motivated by this issue: #669

ykyohei commented on sotodlib/hwp/g3thwp.py:
```diff
@@ -643,18 +643,20 @@ def _set_empty_axes(self, aman):
         aman = sotodlib.core.AxisManager(tod.dets, tod.samps)
         self._set_empty_axes(aman)
         aman.timestamps[:] = tod.timestamps
+        if err: aman.hwp_err[:] = np.ones(len(tod.timestamps))
```
I prefer something like
`aman.hwp_err[:] = np.full_like(tod.timestamps, err, dtype=bool)`

hwp_err is misleading as hwp angle uncertainty, so should we change to err_flag or err_msg?

But I think the error flag can be just a single bool value not an array.
Can we write a single bool value error flag to the h5 metadata file instead?
Or saving simple message like "No data", "data is too short", "Error in hwp angle calculation" would be more helpful?

ykyohei commented on sotodlib/hwp/g3thwp.py:
```diff
@@ -643,18 +643,20 @@ def _set_empty_axes(self, aman):
-    def _write_empty_solution_h5(self, tod, output=None, h5_address=None):
-
+    def _write_empty_solution_h5(self, tod, output=None, h5_address=None, err=False):
+        
```
Please delete unnecessary spaces. There is another one.

yukisakurai commented:
Changed from boolean array to single logger message, `hwp_logger`.
The same stuff is happening with the `eval` flag, so I renamed it to `hwp_angle_ver2_flag` and changed to a single boolean.

ykyohei commented on sotodlib/site_pipeline/make_hwp_solutions.py:
```diff
@@ -18,6 +18,7 @@
 from sotodlib.hwp.g3thwp import G3tHWP
 from sotodlib.site_pipeline import util
 default_logger = util.init_logger(__name__, 'make-hwp-solutions: ')
+print(sotodlib.hwp.g3thwp.__file__)
```
This line looks unnecessary.

ykyohei commented on sotodlib/hwp/g3thwp.py:
```diff
@@ -737,15 +736,19 @@ def write_solution_h5(self, tod, output=None, h5_address=None):
             self._write_empty_solution_h5(tod, output, h5_address)
             return
-
+        
```
unnecessary space

ykyohei commented on sotodlib/hwp/g3thwp.py:
```diff
@@ -723,11 +720,13 @@ def write_solution_h5(self, tod, output=None, h5_address=None):
             data = self.load_data(start, end)
         except Exception as e:
             logger.error(f"Exception '{e}' thrown while loading HWP data. The specified encoder field is missing.")
+            self._write_solution_h5_logger = 'HWP data too short'
```
This logger should be "Error in loading HWPdata"?

yukisakurai replied:
This is not an error, just a short data length and no need to debug, so I prefer to leave it.

ykyohei reviewed (commented):
Thank you. Overall looks good to me. Made some minor comments.
